### ISSUE TITLE
inperfect straightening at the edge

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#513]
data: **errsm_42/t2**

command:
~~~
sct_straighten_spinalcord -i t2.nii.gz -c t2_seg.nii.gz -r 0 -v 2
~~~

result:
![screen shot 2015-09-18 at 12 07 32](https://cloud.githubusercontent.com/assets/2482071/9964405/114a0ade-5dfe-11e5-9e63-4a456e34a2b7.png)

needs more padding?

**Comment by benjamindeleener** [comment]
Have you looked at hanning fitting results?

I noticed some issues with mirroring on another subject (london from Simon I think).

**Comment by jcohenadad** [comment]
well spotted-- see below. I'll look into it...
![screen shot 2015-10-14 at 23 19 05](https://cloud.githubusercontent.com/assets/2482071/10504032/13f94764-72ca-11e5-945e-46043b208713.png)

**Comment by jcohenadad** [comment]
caused by discontinuous segmentation at edge-- see below.
![screen shot 2015-10-14 at 23 22 50](https://cloud.githubusercontent.com/assets/2482071/10504063/973ba4f0-72ca-11e5-8143-0121befbb0bf.png)
 
Any suggestions?

**Comment by jcohenadad** [comment]
when using ``robust=1`` in **sct_straighten_spinalcord**:
![screen shot 2015-10-15 at 08 52 43](https://cloud.githubusercontent.com/assets/2482071/10514186/62ec5816-731a-11e5-9e8c-ab803307223f.png)
Many values are wrong (e.g., first ones are all equal to 26, which is wrong). However results look good:
<img width="478" alt="screen shot 2015-10-15 at 08 56 15" src="https://cloud.githubusercontent.com/assets/2482071/10514237/9f5b4f6e-731a-11e5-913f-501e99f61895.png">

**Comment by jcohenadad** [comment]
extensive testing:
~~~
# ====================================================================================
# 2015-10-15 09.45
# branch: master
# ====================================================================================
          dist_max       mse      status
subject                                 
errsm_02  1.610000  0.790000    0.000000
errsm_03  1.360000  0.850000    0.000000
errsm_04       NaN       NaN  200.000000
errsm_05  1.220000  0.620000    0.000000
errsm_09  1.330000  0.820000    0.000000
errsm_10  1.070000  0.690000    0.000000
errsm_11  1.220000  0.640000    0.000000
errsm_12  1.300000  0.710000    0.000000
errsm_13  1.350000  0.830000    0.000000
errsm_14  1.230000  0.640000    0.000000
errsm_16  1.230000  0.730000    0.000000
errsm_17  1.050000  0.670000    0.000000
errsm_18  1.270000  0.720000    0.000000
errsm_21  1.390000  0.750000    0.000000
errsm_22  1.540000  0.790000    0.000000
errsm_23  1.370000  0.840000    0.000000
errsm_24  1.340000  0.720000    0.000000
errsm_25  1.160000  0.730000    0.000000
errsm_26  1.350000  0.790000    0.000000
errsm_30  1.200000  0.740000    0.000000
errsm_31  1.440000  0.750000    0.000000
errsm_32  1.520000  0.720000    0.000000
errsm_33  1.080000  0.660000    0.000000
errsm_34  1.230000  0.680000    0.000000
errsm_35  1.100000  0.750000    0.000000
errsm_42  4.120000  1.120000   99.000000
sct_002   2.030000  0.860000    0.000000
Mean      1.427308  0.754231    3.807692
STD       0.585543  0.100406   19.415497


# ====================================================================================
# 2015-10-15 10.23
# branch: jca_issue_513
# ====================================================================================
          dist_max       mse      status
subject                                 
errsm_02  3.330000  1.010000    0.000000
errsm_03  6.720000  1.860000   99.000000
errsm_04       NaN       NaN  200.000000
errsm_05  3.000000  1.010000    0.000000
errsm_09  3.760000  1.040000    0.000000
errsm_10  2.800000  0.910000    0.000000
errsm_11  5.050000  1.340000   99.000000
errsm_12  1.300000  0.710000    0.000000
errsm_13  2.430000  0.780000    0.000000
errsm_14  1.050000  0.640000    0.000000
errsm_16  1.230000  0.730000    0.000000
errsm_17  2.790000  0.960000    0.000000
errsm_18  1.270000  0.710000    0.000000
errsm_21  4.800000  1.260000   99.000000
errsm_22  2.180000  0.920000    0.000000
errsm_23  1.370000  0.840000    0.000000
errsm_24  7.740000  1.950000   99.000000
errsm_25  1.190000  0.740000    0.000000
errsm_26  1.610000  0.850000    0.000000
errsm_30  4.810000  1.340000   99.000000
errsm_31  2.790000  0.970000    0.000000
errsm_32  5.310000  1.420000   99.000000
errsm_33  4.560000  1.160000   99.000000
errsm_34  2.240000  0.830000    0.000000
errsm_35  2.510000  0.920000    0.000000
errsm_42  4.450000  1.110000   99.000000
sct_002   2.260000  0.910000    0.000000
Mean      3.175000  1.035385   30.461538
STD       1.788421  0.329815   46.597194
~~~

So ``robust=1`` is not a good option. Next test: remove two last points, then do the mirroring.

**Comment by jcohenadad** [comment]
removing the two last points works pretty well:
~~~
# ====================================================================================
# 2015-10-15 12.13
# branch: jca_issue_513
# ====================================================================================
          dist_max       mse  status
subject
errsm_02  1.670000  0.700000       0
errsm_03  1.360000  0.860000       0
errsm_04       NaN       NaN     200
errsm_05  1.220000  0.620000       0
errsm_09  1.400000  0.710000       0
errsm_10  1.070000  0.700000       0
errsm_11  1.290000  0.640000       0
errsm_12  1.300000  0.700000       0
errsm_13  1.190000  0.630000       0
errsm_14  1.340000  0.670000       0
errsm_16  1.170000  0.720000       0
errsm_17  1.050000  0.670000       0
errsm_18  1.270000  0.700000       0
errsm_21  1.390000  0.740000       0
errsm_22  1.540000  0.810000       0
errsm_23  1.050000  0.680000       0
errsm_24  1.340000  0.720000       0
errsm_25  1.220000  0.720000       0
errsm_26  1.350000  0.800000       0
errsm_30  1.200000  0.750000       0
errsm_31  1.500000  0.740000       0
errsm_32  1.520000  0.780000       0
errsm_33  1.180000  0.680000       0
errsm_34  1.260000  0.710000       0
errsm_35  1.100000  0.760000       0
errsm_42  1.820000  0.770000       0
sct_002   2.260000  0.950000       0
Mean      1.348462  0.728077       0
STD       0.262902  0.071890       0
Passed: 26/27
~~~

This is now the default approach.

**Comment by jcohenadad** [comment]
fixed in commit 26993d51a10ed8367a3b07af294284c2fc6bbffc
